### PR TITLE
Use pre-built version of sentry-cocoa SDK

### PR DESCRIPTION
**Pull request by jamescrosswell**
This is to address difficulties we've been having bumping the sentry-cocoa version in this repo (e.g [here](https://github.com/getsentry/sentry-dotnet/pull/3716#issuecomment-2445376208)). 

In this PR, instead of including the sentry-cocoa repo as a git submodule and building this ourselves, we're downloading the most recent release (similar to what is being done [in the Unity repo](https://github.com/getsentry/sentry-unity/blob/8de84e71c6439d43ece603880a95d4daa15188a5/Directory.Build.targets#L162-L169)).

### CODE REVIEW

File: src/Sentry.Bindings.Cocoa/ApiDefinitions.cs [hunk 1]
```diff
@@ -1635,6 +1658,11 @@ interface SentrySDK
     [Export ("isEnabled")]
     bool IsEnabled { get; }
 
+    // @property (readonly, nonatomic, class) SentryReplayApi * _Nonnull replay;
+    [Static]
+    [Export ("replay")]
+    SentryReplayApi Replay { get; }
```
Comment (bruno-garcia) [inline src/Sentry.Bindings.Cocoa/ApiDefinitions.cs:1664]: woooo!! ❤️

File: src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj [hunk 2]
```diff
@@ -44,18 +45,53 @@
...
-    Inputs="..\..\.git\modules\modules\sentry-cocoa\HEAD" Outputs="..\..\modules\sentry-cocoa\Carthage\.built-from-sha">
-    <MSBuild Projects="$(MSBuildProjectFile)" Targets="_InnerBuildSentryCocoaSDK" Properties="TargetFramework=once" />
+  <!-- Downloads and sets up the Cocoa SDK: dotnet msbuild /t:setupCocoaSDK src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj -->
+  <Target Name="SetupCocoaSDK"
+          Condition="$([MSBuild]::IsOSPlatform('OSX')) And !Exists('$(SentryCocoaCache)Sentry-Dynamic.xcframework')"
+          BeforeTargets="BeforeBuild">
+
+    <PropertyGroup>
+      <PropertiesContent>$([System.IO.File]::ReadAllText("../../modules/sentry-cocoa.properties"))</PropertiesContent>
+      <CocoaVersion>$([System.Text.RegularExpressions.Regex]::Match($(PropertiesContent), 'version\s*=\s*(\d+\.\d+\.\d+)').Groups[1].Value)</CocoaVersion>
```
Comment (bruno-garcia) [inline src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj:55]: this regex will match beta releases but extract only the `n.n.n` and not the `-xyz`

**Review by bruno-garcia — CHANGES_REQUESTED** [review]

File: src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj (hunk 2 above)
Comment (jamescrosswell) [reply src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj]: Ah, nice catch. @bitsandfoxes you'll want to [fix that in the Unity repo](https://github.com/getsentry/sentry-unity/blob/8de84e71c6439d43ece603880a95d4daa15188a5/Directory.Build.targets#L149) as well.

I've gone with dead simple - which is to match any sequence of non-whitespace characters:
https://github.com/getsentry/sentry-dotnet/blob/bbd1f64417731aab4d780afda0902066a0129a11/src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj#L55

**Review by bricefriha — APPROVED** [review]
As far as I'm concerned, 
looks fine!

File: src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj (hunk 2 above)
Comment (jamescrosswell) [reply src/Sentry.Bindings.Cocoa/Sentry.Bindings.Cocoa.csproj]: @bruno-garcia the issue you found in the regex has been addressed. Any other "changes requested" or can we merge?

**Review by bruno-garcia — APPROVED** [review]

**Comment by bruno-garcia** [comment]
I was a big fan of the flexibility of modifying the underlying library, pushing a branch and having a feature cut through that way but I realize it's a lot of overhead for this very rarely used benefit.

**Pull request merged**